src/NumericalAlgorithms/Convergence/HasConverged.cpp

```diff
@@ -84,15 +85,16 @@ std::ostream& operator<<(std::ostream& os,
       case Reason::AbsoluteResidual:
         return os
                << "AbsoluteResidual - The residual magnitude has decreased to "
-               << has_converged.criteria_.absolute_residual << " or below ("
-               << has_converged.residual_magnitude_ << ").";
+               << get_output(has_converged.criteria_.absolute_residual)
```

Comment:
Why are you running these through `get_output`?

Reply:
To print them at a reasonable precision without having to do things like `<< std::setprecision`

Reply:
But `get_output` just prints at the default precision.  If anything, I would expect any stream passed in to be at a higher precision than that.

Reply:
Default precision is good. Without the `get_output` this prints 16(ish) digits.

Reply:
I see that we do indeed set the precision to something ridiculous. Looks good then.